src/constants/currency.ts

```diff
@@ -220,49 +221,61 @@ export const getAvailableMarketNames = memoizeOne(() => {
 		quoteCurrencyKey: CurrencyKey;
 		pair: string;
 	}> = [];
+
 	const excludedSynths = [SYNTHS_MAP.iMKR, SYNTHS_MAP.sMKR];
 	const synths = SYNTHS.filter((synth) => !excludedSynths.includes(synth));
 
-	// fiat markets trade against all synths
-	FIAT_SYNTHS.forEach((quoteCurrencyKey) => {
-		synths
-			// ignore self
-			.filter((baseCurrencyKey) => quoteCurrencyKey !== baseCurrencyKey)
-			.forEach((baseCurrencyKey) => {
-				marketNames.push({
-					baseCurrencyKey,
-					quoteCurrencyKey,
-					pair: toMarketPair(baseCurrencyKey, quoteCurrencyKey),
-				});
-			});
-	});
-
-	// Each iteration a crypto synth is added to be skipped in the next one
-	// So for [sBTC, sETH] crypto pairs, we would only end up with sETH/sBTC
-	const skipCryptoQuotes: CurrencyKeys = [];
-
-	// crypto markets trade against all synths (ex fiat, ex existing crypto market)
-	CRYPTO_SYNTHS_BY_MC.forEach((quoteCurrencyKey: CurrencyKey) => {
-		synths
-			.filter(
-				(baseCurrencyKey: CurrencyKey) =>
-					quoteCurrencyKey !== baseCurrencyKey &&
-					![...FIAT_SYNTHS, ...skipCryptoQuotes].includes(baseCurrencyKey)
-			)
-			.forEach((baseCurrencyKey: CurrencyKey) => {
+	for (let i = 0; i < synths.length; i++) {
+		const currencyA = SYNTHS[i];
+		for (let j = 0; j < synths.length; j++) {
+			const currencyB = SYNTHS[j];
+			if (currencyA !== currencyB) {
 				marketNames.push({
-					baseCurrencyKey,
-					quoteCurrencyKey,
-					pair: toMarketPair(baseCurrencyKey, quoteCurrencyKey),
+					baseCurrencyKey: currencyA,
+					quoteCurrencyKey: currencyB,
+					pair: toMarketPair(currencyA, currencyB),
 				});
-			});
-
-		skipCryptoQuotes.push(quoteCurrencyKey);
-	});
+			}
+		}
+	}
 
 	return marketNames;
 });
 
 export const getFilteredMarketNames = memoizeOne((quoteCurrencyKey: CurrencyKey) =>
 	getAvailableMarketNames().filter((marketName) => marketName.quoteCurrencyKey === quoteCurrencyKey)
 );
+
+// sBTC / sUSD
+// sUSD / sBTC
+
+export const getMarketPairByMC = memoizeOne(
+	(baseCurrencyKey: CurrencyKey, quoteCurrencyKey: CurrencyKey) => {
+		const marketPair = {
+			base: baseCurrencyKey,
+			quote: quoteCurrencyKey,
+			reversed: false,
+		};
+
+		const marketPairReversed = {
+			base: quoteCurrencyKey,
+			quote: baseCurrencyKey,
+			reversed: true,
+		};
+
+		// handle fiat first - it must always be the quote
+		if (FIAT_SYNTHS.includes(quoteCurrencyKey)) {
+			return marketPair;
```

Comment:
we could rank fiat synths here too... but for now, I left it unsorted.

Reply:
agreed